src/libraries/System.Private.Xml/src/System/Xml/Core/XmlTextWriter.cs

```diff
@@ -981,6 +984,117 @@ public override void WriteBinHex(byte[] buffer, int index, int count)
             }
         }
 
+        // Override in order to handle Xml simple typed values and to pass resolver for QName values
+        public override void WriteValue(object value)
```

Comment:
Whole logic for writing with buffer is duplicated in XmlTextWriter and XmlRawWriter.
One suggestion is to create "public" class( this is needed as XmlTextWriter is public) and encapsulate logic in it.
The other is to have some static helper class which works by ref with char[].

Which one do you prefer ?

Reply:
Duplication logic reduced, but the question still stays open.